spring-cloud-aws-parameter-store-config/src/main/java/org/springframework/cloud/aws/paramstore/AwsParamStorePropertySourceLocator.java

```diff
@@ -85,9 +86,11 @@ public AwsParamStorePropertySourceLocator(AWSSimpleSystemsManagement ssmClient,
 		this.contexts.add(defaultContext + "/");
 		addProfiles(this.contexts, defaultContext, profiles);
 
-		String baseContext = prefix + "/" + appName;
-		this.contexts.add(baseContext + "/");
-		addProfiles(this.contexts, baseContext, profiles);
+		if (!this.properties.getDefaultContext().equals(appName)) {
```

Comment:
Have you considered changing `this.context` from `List` to `TreeSet` so that it guarantees element uniqueness without adding any `ifs` later in the code?

Reply:
Will change.